File: backend/src/main/java/com/joying/chat/dto/PushSubscriptionRequest.java
package com.joying.chat.dto;

import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * 웹 푸시 구독 등록 요청.
 */
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class PushSubscriptionRequest {

	@NotBlank(message = "엔드포인트는 필수입니다")
	private String endpoint;

	@NotBlank(message = "p256dh 키는 필수입니다")
	private String p256dh;

	@NotBlank(message = "auth 키는 필수입니다")
	private String auth;

	private String userAgent;
}

File: backend/src/main/kotlin/com/joying/chat/dto/PushSubscriptionRequest.kt
package com.joying.chat.dto

import jakarta.validation.constraints.NotBlank

/**
 * Web Push 구독 등록 요청
 */
data class PushSubscriptionRequest(
    @field:NotBlank(message = "엔드포인트는 필수입니다")
    val endpoint: String,

    @field:NotBlank(message = "p256dh 키는 필수입니다")
    val p256dh: String,

    @field:NotBlank(message = "auth 키는 필수입니다")
    val auth: String,

    val userAgent: String? = null
)

/**
 * Web Push 구독 해제 요청
 */
data class PushUnsubscribeRequest(
    @field:NotBlank(message = "엔드포인트는 필수입니다")
    val endpoint: String
)